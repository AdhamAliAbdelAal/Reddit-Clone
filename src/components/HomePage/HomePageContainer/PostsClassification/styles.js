import { Box, Button } from '@mui/material';
import { styled } from '@mui/system';
import { Link } from 'react-router-dom';

export const PostsClassContainer = styled('div')(() => ({
  width: '100%',
  margin: '16px 0',
  backgroundColor: '#fff',
  padding: '14px 12px',
  borderRadius: 4,
  display: 'flex',
  gap: 10,
  border: '1px solid #ccc',
}));

export const CustomClassLink = styled(Link)(() => ({
  textDecoration: 'none',
  position: 'relative',
}));

export const PostClassesList = styled(Box)(() => ({
  position: 'absolute',
  top: '100%',
  left: 0,
  backgroundColor: '#fff',
  borderRadius: 4,
  display: 'flex',
}));

export const PostClassSmButton = styled(Link)(({ active, theme }) => ({
  padding: 8,
  fontWeight: 400,
  fontSize: 14,
  textDecoration: 'none',
  display: 'flex',
  textTransform: 'capitalize',
  gap: 4,
  justifyContent: 'center',
  alignItems: 'center',
  color: (active ? theme.palette.primary.main : theme.palette.third.main),
  '&:hover': {
    color: (active ? theme.palette.primary.main : '#000'),
    backgroundColor: '#e9f5fd',
  },
}));

export const ClassButton = styled(Button)(({ active }) => ({
  display: 'flex',
  gap: 6,
  borderRadius: 20,
  textTransform: 'capitalize',
  minWidth: 'auto',
  padding: '4px 10px',
  fontSize: 14,
  fontWeight: 'bolder',
  backgroundColor: (active ? '#f6f7f8' : null),
}));
