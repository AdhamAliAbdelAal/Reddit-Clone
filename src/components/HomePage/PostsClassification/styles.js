import { Button } from '@mui/material';
import { styled } from '@mui/system';
import { Link } from 'react-router-dom';

export const PostsClassContainer = styled('div')(() => ({
  width: '100%',
  margin: '16px 0',
  backgroundColor: '#fff',
  padding: '14px 12px',
  borderRadius: 4,
  display: 'flex',
  gap: 10,
  border: '1px solid #ccc',
}));

export const CustomClassLink = styled(Link)(() => ({
  textDecoration: 'none',
}));

export const ClassButton = styled(Button)(({ active }) => ({
  display: 'flex',
  gap: 6,
  borderRadius: 20,
  textTransform: 'capitalize',
  minWidth: 'auto',
  padding: '4px 10px',
  fontSize: 14,
  fontWeight: 'bolder',
  backgroundColor: (active ? '#f6f7f8' : null),
}));
