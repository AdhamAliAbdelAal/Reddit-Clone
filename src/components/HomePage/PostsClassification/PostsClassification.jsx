import RocketOutlinedIcon from '@mui/icons-material/RocketOutlined';
import RocketIcon from '@mui/icons-material/Rocket';
import LocalFireDepartmentOutlinedIcon from '@mui/icons-material/LocalFireDepartmentOutlined';
import LocalFireDepartmentIcon from '@mui/icons-material/LocalFireDepartment';
import NewReleasesOutlinedIcon from '@mui/icons-material/NewReleasesOutlined';
import NewReleasesIcon from '@mui/icons-material/NewReleases';
import UploadOutlinedIcon from '@mui/icons-material/UploadOutlined';
import UploadIcon from '@mui/icons-material/Upload';
import { useParams } from 'react-router-dom';
import { useEffect, useState } from 'react';
import { CustomClassLink, PostsClassContainer, ClassButton } from './styles';

function PostsClassification() {
  const { postClass } = useParams();
  const [activeClass, setActiveClass] = useState({
    best: (postClass === 'best' || !postClass),
    hot: (postClass === 'hot'),
    new: (postClass === 'new'),
    top: (postClass === 'top'),
  });
  useEffect(() => {
    setActiveClass({
      best: (postClass === 'best' || !postClass),
      hot: (postClass === 'hot'),
      new: (postClass === 'new'),
      top: (postClass === 'top'),
    });
  }, [postClass]);
  return (
    <PostsClassContainer>
      <CustomClassLink to="/best/">
        <ClassButton color={(activeClass.best ? 'primary' : 'third')} active={activeClass.best}>
          {(activeClass.best ? <RocketIcon /> : <RocketOutlinedIcon />)}
          <div>
            best
          </div>
        </ClassButton>
      </CustomClassLink>
      <CustomClassLink to="/hot/">
        <ClassButton color={(activeClass.hot ? 'primary' : 'third')} active={activeClass.hot}>
          {(activeClass.hot ? <LocalFireDepartmentIcon /> : <LocalFireDepartmentOutlinedIcon />)}
          <div>
            hot
          </div>
        </ClassButton>
      </CustomClassLink>
      <CustomClassLink to="/new/">
        <ClassButton color={(activeClass.new ? 'primary' : 'third')} active={activeClass.new}>
          {(activeClass.new ? <NewReleasesIcon /> : <NewReleasesOutlinedIcon />)}
          <div>
            new
          </div>
        </ClassButton>
      </CustomClassLink>
      <CustomClassLink to="/top/">
        <ClassButton color={(activeClass.top ? 'primary' : 'third')} active={activeClass.top}>
          {(activeClass.top ? <UploadIcon /> : <UploadOutlinedIcon />)}
          <div>
            top
          </div>
        </ClassButton>
      </CustomClassLink>
    </PostsClassContainer>
  );
}

export default PostsClassification;
