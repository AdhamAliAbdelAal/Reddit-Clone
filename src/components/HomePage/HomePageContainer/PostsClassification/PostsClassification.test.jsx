import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import PostsClassification from './PostsClassification';

describe(PostsClassification, () => {
  it('PostsClassification renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <PostsClassification />
          ,
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
