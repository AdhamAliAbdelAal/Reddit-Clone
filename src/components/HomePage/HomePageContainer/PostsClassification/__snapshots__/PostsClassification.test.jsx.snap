// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsClassification PostsClassification renders correctly 1`] = `
Array [
  <div
    className="css-1i4l6gc"
  />,
  ",",
]
`;
